### ISSUE TITLE
Why two "separate" MARS values in current_law_policy.json ?

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#1267]
A number of policy parameters are indexed by filing unit type (`MARS`).  All those parameters have the following infomation in the `current_law_policy.json` file:
```
        "col_var": "MARS",
        "col_label": ["single", "joint", "separate", "head of household", "widow", "separate"],
```
When you look at the values for the third and sixth filing unit types (both of which are labeled "separate") you see that they are exactly the same.  So, why do we need the sixth filing unit type (that is, the second "separate")?

It seems like the sixth type is redundant and can be eliminated.  Or have I missed something?

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @GoFroggyRun @codykallen

**Comment by martinholmer** [comment]
With respect to issue #1267, the  current version of the `puf.csv` file contains `MARS` values that range from 1 through 4.

There are no records with MARS==5 or MARS==6.  Here are the unweighted counts by MARS value:
```
$ awk -F, 'NR>1{n[$8]++}END{for(t in n)print t,n[t]}' puf.csv 
1   85765
2  109265
3    4289
4   20495
```

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @GoFroggyRun @codykallen

**Comment by feenberg** [comment]
On Wed, 29 Mar 2017, Martin Holmer wrote:

> 
> A number of policy parameters are indexed by filing unit type (MARS). All
> those parameters have the following infomation in the
> current_law_policy.json file:
>
>         "col_var": "MARS",
>         "col_label": ["single", "joint", "separate", "head of household", "w
> idow", "separate"],
> 
> When you look at the values for the third and sixth filing unit types (both
> of which are labeled "separate") you see that they are exactly the same. So,
> why do we need the sixth filing unit type (that is, the second "separate")?
> 
> It seems like the sixth type is redundant and can be eliminated. Or have I
> missed something?

The PUF comes with the two different "separates" but as you say, it never 
affects tax liability. In taxsim we keep the distinction in the code but 
it would make sense to consolidate the two.

dan

**Comment by codykallen** [comment]
@martinholmer, I have never seen a reason for having MARS==6 (redundant 'separate' entry). I agree with @feenberg that they should be consolidated.

I can understand having a space for MARS==5 in case there are widows in any PUF, but those could be reclassified as married filing jointly for purposes of calculating taxes.

**Comment by martinholmer** [comment]
@feenberg and @codykallen,  Thanks to you both for the quick feedback on MARS issue #1267.

I've looked at two more things since my original post.  

First, the Form 1040 has MARS values of 1-5.  

Second, the IRS-SOI PUF documentation has MARS values of 1-4 (apart from the "aggregate" record that has MARS==0) because units with MARS==5 (widows) are assigned a MARS value of 3 (separate) in the PUF even though those two groups are not always treated the same under current-law policy.  Maybe this is a privacy issue.

So, we seem to have a consensus.  I'll prepare a pull request that eliminates the second "separate" category (MARS==6) in the `current_law_policy.json` file and add to the input reading logic a check to make sure that all the MARS values in the input data are in the [1,5] range.

**Comment by martinholmer** [comment]
Issue #1267 has been resolved by merged pull request #1268.
